File: AdisyonUygulamaJavaXMLsiz/app/src/main/java/com/example/adisyonuygulamajava/model/Urun.java
package com.example.adisyonuygulamajava.model;

public class Urun {

    private int id;
    private String urun_ad;
    private float urun_fiyat;
    private String urun_resim;
    private int urun_adet;
    private Kategori urunKategori;

    // Constructor
    public Urun(int id, String urun_ad, float urun_fiyat, String urun_resim, int urun_adet, Kategori urunKategori) {
        this.id = id;
        this.urun_ad = urun_ad;
        this.urun_fiyat = urun_fiyat;
        this.urun_resim = urun_resim;
        this.urun_adet = urun_adet;
        this.urunKategori = urunKategori;
    }

    // Getters and Setters
    public int getId() {
        return id;
    }

    public String getUrun_ad() {
        return urun_ad;
    }

    public void setUrun_ad(String urun_ad) {
        this.urun_ad = urun_ad;
    }

    public float getUrun_fiyat() {
        return urun_fiyat;
    }

    public void setUrun_fiyat(float urun_fiyat) {
        this.urun_fiyat = urun_fiyat;
    }

    public String getUrun_resim() {
        return urun_resim;
    }

    public void setUrun_resim(String urun_resim) {
        this.urun_resim = urun_resim;
    }

    public int getUrun_adet() {
        return urun_adet;
    }

    public void setUrun_adet(int urun_adet) {
        this.urun_adet = urun_adet;
    }

    public Kategori getUrunKategori() {
        return urunKategori;
    }

    public void setUrunKategori(Kategori urunKategori) {
        this.urunKategori = urunKategori;
    }
}

File: adisyon_flutter/lib/model/Urun.dart
import 'Kategori.dart';

class Urun {
  final int id;
  String urunAd;
  double urunFiyat;
  String urunResim;
  int urunAdet;
  Kategori urunKategori;

  Urun({
    required this.id,
    required this.urunAd,
    required this.urunFiyat,
    required this.urunResim,
    required this.urunAdet,
    required this.urunKategori,
  });

  factory Urun.fromJson(Map<String, dynamic> json) => Urun(
    id: json['id'],
    urunAd: json['urun_ad'],
    urunFiyat: (json['urun_fiyat'] as num).toDouble(),
    urunResim: json['urun_resim'],
    urunAdet: json['urun_adet'],
    urunKategori: Kategori.fromJson(json['urunKategori']),
  );

  Map<String, dynamic> toJson() => {
    'id': id,
    'urun_ad': urunAd,
    'urun_fiyat': urunFiyat,
    'urun_resim': urunResim,
    'urun_adet': urunAdet,
    'urunKategori': urunKategori.toJson(),
  };
}
